### PR TITLE
Update Readmes for TestProxy CLI

### PR DESCRIPTION
**Pull request by JimSuplizio**
Update the Azure.Sdk.Tools.TestProxy/README.md and documentation/asset-sync/README.md to accurately reflect the CLI commands and start arguments.

### CODE REVIEW

File: tools/test-proxy/documentation/asset-sync/README.md [hunk 1]
```diff
@@ -25,49 +25,61 @@ An `assets.json` takes the form:
...
 Interactions with the external assets repository are accessible when the proxy is actively serving requests. These are available through routes:
 
 | Route | Description |
 |---|---|
-| `/Playback/Restore` | Retrieve files from external git repo as targeted in the SHA from assets.json |
-| `/Playback/Reset` | Discard pending changes and reset to the original SHA from targeted assets.json. |
-| `/Record/Push` | Push changes if they are pending for files described by targeted assets.json. |
+| `/Playback/Restore` | Retrieve files from external git repo as targeted in the Tag from assets.json |
+| `/Playback/Reset` | Discard pending changes and reset to the original Tag from targeted assets.json. |
+| `/Record/Push` | Push changes if they are pending for files described by targeted assets.json. Updates the Tag in assets.json the new tag generated from the push operation. |
```
Comment (scbedd) [inline tools/test-proxy/documentation/asset-sync/README.md:53]:
```suggestion
| `/Record/Push` | Push changes if they are pending for files described by targeted assets.json. Updates the Tag in assets.json to match the new tag generated from the push operation. |
```

File: tools/test-proxy/documentation/asset-sync/README.md [hunk 2]
```diff
@@ -25,49 +25,61 @@ An `assets.json` takes the form:
...
+## test-proxy CLI commands
 
-The test-proxy also offers interactions with the external assets repository as a CLI app. What this means is that one could invoke
+The test-proxy also offers interactions with the external assets repository as a CLI. Invoking `test-proxy --help` will show the available list of commands. `test-proxy <command> --help` will show the help and options for an individual command. The options for a given command are all `--<option>`, for example, `--assets-json-path`, but each option has an abbreviation shown in the help, those are a single dash. For example the abbreviation for `--assets-json-path` is `-a`.
+
+### The following CLI commands are available for manipulation of assets
+
+#### Restore
+
+Restore will pull the assets at the Tag, from the AssetsRepo referenced by the targeted assets.json
```
Comment (scbedd) [inline tools/test-proxy/documentation/asset-sync/README.md:63]:
```suggestion
A restore operation is merely a test-proxy-encapsulated `clone or pull` operation. A given `asets.json` provides the target `Tag` and `AssetsRepo`.
```

I don't like this really either, but I don't like what it is now.

File: tools/test-proxy/documentation/asset-sync/README.md [hunk 3]
```diff
@@ -25,49 +25,61 @@ An `assets.json` takes the form:
...
-> test-proxy --command restore --asetsJsonPath <assetsJsonPath>
+> test-proxy restore --assets-json-path <assetsJsonPath>
 ```
 
-to pull the necessary recordings files down for a targeted assets.json. The following commands are available.
+#### Reset
+
+Reset will reset the assets back to the Tag, from the AssetsRepo referenced by the targeted assets.json. Reset would be used if the assets were already restored, modified (maybe re-recorded while library development was done), and then needed to be reset back to their original files. If there are pending changes, the user will be prompted to overwrite. If there are no pending changes, then reset is no-op, otherwise, the following prompt will be displayed.
+`There are pending git changes, are you sure you want to reset? [Y|N]`
+Selecting N will leave things as they are, selecting Y will restore the assets back to their original files referenced by the Tag, from the AssetsRepo referenced by the targeted assets.json file.
```
Comment (scbedd) [inline tools/test-proxy/documentation/asset-sync/README.md:73]:
```suggestion

- Selecting N will leave things as they are. 
- Selecting Y will discard pending changes and reset the locally cloned assets to the Tag within the targeted `assets.json`.
```

**Review by scbedd — COMMENTED** [review]
There a couple sentences that I would prefer to be reworded, as they read very awkwardly.

Other than those couple sentences this is excellent!

File: tools/test-proxy/documentation/asset-sync/README.md (hunk 1 above)
Comment (JimSuplizio) [reply tools/test-proxy/documentation/asset-sync/README.md:53]: I'll update with the next commit.

File: tools/test-proxy/documentation/asset-sync/README.md (hunk 2 above)
Comment (JimSuplizio) [reply tools/test-proxy/documentation/asset-sync/README.md:63]: I'll update with the next commit.

File: tools/test-proxy/documentation/asset-sync/README.md [hunk 4]
```diff
@@ -25,49 +25,61 @@ An `assets.json` takes the form:
...
 
 ```bash
-> test-proxy --command restore --asetsJsonPath <assetsJsonPath>
+> test-proxy restore --assets-json-path <assetsJsonPath>
 ```
 
-to pull the necessary recordings files down for a targeted assets.json. The following commands are available.
+#### Reset
+
+Reset will reset the assets back to the Tag, from the AssetsRepo referenced by the targeted assets.json. Reset would be used if the assets were already restored, modified (maybe re-recorded while library development was done), and then needed to be reset back to their original files. If there are pending changes, the user will be prompted to overwrite. If there are no pending changes, then reset is no-op, otherwise, the following prompt will be displayed.
```
[the remark this answers is not in the document]
Comment (JimSuplizio) [reply tools/test-proxy/documentation/asset-sync/README.md:71]: I'll update with the next commit.

File: tools/test-proxy/documentation/asset-sync/README.md (hunk 3 above)
Comment (JimSuplizio) [reply tools/test-proxy/documentation/asset-sync/README.md:73]: I'll update with the next commit.

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md [hunk 5]
```diff
@@ -174,10 +175,12 @@ $env:ASPNETCORE_URLS="http://*3331;https://*:8881"  // set custom ports for both
 
 #### Input arguments
 
-Use command line argument `--urls` to bind to a non-default host and port. This configuration will override the environment configuration.  For example, to only bind to localhost http 5000, provide the argument `--urls http://localhost:9000`.
+When starting the test proxy there are Verbs, as shown in the test-proxy --help output above. For the start  verb, there can be additional command arguments that need to get passed to **Host.CreateDefaultBuilder** 'as is'. The way this is done is through the use of `--` or dashdash as it's referred to. If `--` is on the command line, everything after will be treated as arguments that will be passed into this Host call.
+
+For example, you can use command line argument `--urls` to bind to a non-default host and port. This configuration will override the environment configuration. To bind to localhost http 5000, provide the argument `-- --urls http://localhost:9000`. Note that `--`, space, then the `--urls http://localhost:9000`.
```
Comment (benbp) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md:180]: Should this say `localhost http 9000` instead of 5000?

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md [hunk 6]
```diff
@@ -174,10 +175,12 @@ $env:ASPNETCORE_URLS="http://*3331;https://*:8881"  // set custom ports for both
...
 
 ```powershell
-test-proxy --urls "http://localhost:9000;https://localhost:9001"
+test-proxy -- --urls "http://localhost:9000;https://localhost:9001"
```
Comment (benbp) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md:183]: To me this feels confusing/un-intuitive. Is it only `urls` that needs to get passed to the inner host builder? If we control the outer cli wrapper, why is this necessary?

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md (hunk 5 above)
Comment (JimSuplizio) [reply tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md:180]: Yes, good catch, I'll fix with next commit.

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md (hunk 6 above)
Comment (JimSuplizio) [reply tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md:183]: @benbp, there's a whole bunch of random items that can be passed to Host.CreateDefaultBuilder. Normally, this is done by just passing args directly to the Host.CreateDefaultBuilder(args) call. Because we're using CommandLineParser and the command line, itself, consists of Verbs and their arguments, the only way to get the "everything else to pass to Host.CreateDefaultBuilder" was to use CommandLineParser's dashdash option which says "grab the rest of the args as is". This is only valid for the [StartOptions](https://github.com/Azure/azure-sdk-tools/blob/main/tools/test-proxy/Azure.Sdk.Tools.TestProxy/CommandParserOptions/StartOptions.cs#L15).

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md [hunk 7]
```diff
@@ -174,10 +175,12 @@ $env:ASPNETCORE_URLS="http://*3331;https://*:8881"  // set custom ports for both
 
 #### Input arguments
 
-Use command line argument `--urls` to bind to a non-default host and port. This configuration will override the environment configuration.  For example, to only bind to localhost http 5000, provide the argument `--urls http://localhost:9000`.
+When starting the test proxy there are Verbs, as shown in the test-proxy --help output above. For the start  verb, there can be additional command arguments that need to get passed to **Host.CreateDefaultBuilder** 'as is'. The way this is done is through the use of `--` or dashdash as it's referred to. If `--` is on the command line, everything after will be treated as arguments that will be passed into this Host call.
```
[the remark this answers is not in the document]
Comment (JimSuplizio) [reply tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md:178]: I'll update with the next commit.

**Review by benbp — APPROVED** [review]

**Comment by scbedd** [comment]
Resolves #3698
